firstvoices/backend/models/signals/mtd_signals.py

```diff
@@ -0,0 +1,47 @@
+from django.db import transaction
+from django.db.models import Count
+from django.db.models.signals import m2m_changed, post_delete, post_save
+from django.dispatch import receiver
+
+from backend.models import Category, DictionaryEntry
+from backend.models.constants import Visibility
+from backend.tasks.build_mtd_export_format import build_index_and_calculate_scores
+from firstvoices.celery import link_error_handler
+
+
+def rebuild_mtd_index(site_slug):
+    transaction.on_commit(
+        lambda: build_index_and_calculate_scores.apply_async(
+            (site_slug,),
+            link_error=link_error_handler.s(),
+        )
+    )
+
+
+@receiver(post_save, sender=DictionaryEntry)
+@receiver(post_delete, sender=DictionaryEntry)
+def request_update_mtd_index(sender, instance, **kwargs):
+    if instance.visibility == Visibility.PUBLIC:
```

Comment:
Does this handle cases where an entry is changed from public to members and vice versa?

Reply:
Good catch, it previously did not. I have now added pre_save and modified the post_save signal to check the previous visibility and rebuild index on the basis of that. That should cover the public to non-public visibility cases.